### PR TITLE
q1

### PR DESCRIPTION
**Pull request by Adrianbarros**
question 1 done

### CODE REVIEW

File: AdrianBarros/quiz/q1.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+time complexity O(n+n) = O(n) two Separate loops but we only go over the whole loop once
+space complexity: O(1)
+
+edge cases: 
+not as much as an edge case as a follow up question is what if we add neg numbers in the array:
+        then we change max from 0 to -inf
+what is if the array is empty or the number is 0 or neg
+            return 0
+'''
```
Comment (knrodriguez) [inline AdrianBarros/quiz/q1.py:24]: Good approach! Nice job adding edge cases.

File: AdrianBarros/quiz/q1.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Given an array of positive numbers and a positive number k, find the maximum sum of any contiguous subarray of size k.
+'''
+
+'''
+input: unsorted array + int
+output: max sum
+step1: start declaring two pointers
+step2: loop through the array adding all the k numbers
+step3: compare with the last max number and keep going
+setp4: substract the first pointer to it and move both pointers
```
Comment (knrodriguez) [inline AdrianBarros/quiz/q1.py:12]: I think you mean to say subtract the first pointer's value.

File: AdrianBarros/quiz/q1.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    while ptr2 < len(arr):
+        curr_sum -= arr[ptr1]
+        ptr1 += 1
+        curr_sum += arr[ptr2]
+        max_sum = max(max_sum, curr_sum)
+        ptr2 += 1
+    return max_sum
+
+
```
Comment (knrodriguez) [inline AdrianBarros/quiz/q1.py:47]: Missing test case that asserts the edge cases you wrote above.

**Comment by knrodriguez** [comment]
Make sure you write a test case that will satisfy your edge case. Otherwise, nice work! :+1:

**Comment by knrodriguez** [comment]
Great approach and solution! Don't forget to add a test case to assert your edge case.
